library/recommend.cc

```diff
@@ -141,22 +141,28 @@ int main(int argc, char* argv[])
   }
 
   FILE* fB;
+  int errnofb;
   FILE* fU;
+  int errnofu;
   FILE* fI;
+  int errnofi;
 
-  if ((fB = fopen(blacklistfilename.c_str(), "r")) == NULL)
+  VW::file_open(&fB, blacklistfilename.c_str(), "r", &errnofb);
+  if (errnofb != 0)
   {
     fprintf(stderr, "can't open %s: %s\n", blacklistfilename.c_str(), strerror(errno));
```

Comment:
@jackgerrits should this be using VW::strerror_to_string?

Reply:
not relevant to this pr though

Reply:
Replaced.